### PR TITLE
[SYCL] Add test for set_arg with local_accessor

### PR DESCRIPTION
**Pull request by AlexeySachkov**

### CODE REVIEW

**Comment by AlexeySachkov** [comment]
Corresponding product change: intel/llvm#7313

File: SYCL/Regression/set-arg-local-accessor.cpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+kernel getKernel(const queue &Q) {
+  cl_int status;
+
+  auto device_cl = get_native<backend::opencl>(Q.get_device());
+  auto context_cl = get_native<backend::opencl>(Q.get_context());
+  auto program_cl =
+      clCreateProgramWithSource(context_cl, 1, &clSource, nullptr, &status);
+  handle(status);
```
Comment (romanovvlad) [inline SYCL/Regression/set-arg-local-accessor.cpp]:
```suggestion
  assert(CL_SUCCESS == status);
```

File: SYCL/Regression/set-arg-local-accessor.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+using namespace sycl;
+
+const char *clSource = R"(
+    kernel void test(global int *a, local float *b, int n) {
+        if (get_local_id(0) != 0)
+            return;
+        for (int i = 0; i < n; i++)
+            b[i] = i;
+        barrier(CLK_LOCAL_MEM_FENCE);
```
Comment (romanovvlad) [inline SYCL/Regression/set-arg-local-accessor.cpp:18]: Could you please clarify why the `barrier` is required here?

File: SYCL/Regression/set-arg-local-accessor.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    cgh.set_arg(0, acc_global);
+    cgh.set_arg(1, acc_local);
+    cgh.set_arg(2, N_slm);
+
+    cgh.parallel_for(nd_range<1>(N_wg, 1), K);
+  });
+
+  auto acc_global = b.get_access<access::mode::read>();
+  for (int i = 0; i < N_wg; i++) {
+    if (acc_global[i] != 1.0f) {
```
Comment (romanovvlad) [inline SYCL/Regression/set-arg-local-accessor.cpp]:
```suggestion
    if (acc_global[i] != 1) {
```
?

File: SYCL/Regression/set-arg-local-accessor.cpp [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+int main() {
+  queue Q;
+  auto K = getKernel(Q);
+
+  constexpr cl_int N_slm = 256;
+  constexpr int N_wg = 32;
+
+  cl_int init[N_wg] = {-1};
```
Comment (romanovvlad) [inline SYCL/Regression/set-arg-local-accessor.cpp]: Could you please clarify why the first element needs to be initialized to `-1`?

File: SYCL/Regression/set-arg-local-accessor.cpp [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    }
+)";
+
+void handle(cl_int status) {
+  if (status != CL_SUCCESS)
+    throw std::runtime_error("OpenCL error occurred");
+}
+
+kernel getKernel(const queue &Q) {
+  cl_int status;
```
Comment (romanovvlad) [inline SYCL/Regression/set-arg-local-accessor.cpp]:
```suggestion
  cl_int status = CL_SUCCESS;
```

File: SYCL/Regression/set-arg-local-accessor.cpp [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    cgh.set_arg(1, acc_local);
+    cgh.set_arg(2, N_slm);
+
+    cgh.parallel_for(nd_range<1>(N_wg, 1), K);
+  });
+
+  auto acc_global = b.get_access<access::mode::read>();
+  for (int i = 0; i < N_wg; i++) {
+    if (acc_global[i] != 1.0f) {
+      std::cout << "Error in WG " << i << std::endl;
```
Comment (romanovvlad) [inline SYCL/Regression/set-arg-local-accessor.cpp]:
```suggestion
      std::cerr << "Error in WG " << i << std::endl;
```

File: SYCL/Regression/set-arg-local-accessor.cpp (hunk 1 above)
Comment (AlexeySachkov) [reply SYCL/Regression/set-arg-local-accessor.cpp]: Done in 8cfb6f4c98e7f7d71f6d7f2da3d8bb37e0496759

File: SYCL/Regression/set-arg-local-accessor.cpp (hunk 3 above)
Comment (AlexeySachkov) [reply SYCL/Regression/set-arg-local-accessor.cpp]: Done in 8cfb6f4c98e7f7d71f6d7f2da3d8bb37e0496759

File: SYCL/Regression/set-arg-local-accessor.cpp (hunk 6 above)
Comment (AlexeySachkov) [reply SYCL/Regression/set-arg-local-accessor.cpp]: Done in 8cfb6f4c98e7f7d71f6d7f2da3d8bb37e0496759

File: SYCL/Regression/set-arg-local-accessor.cpp (hunk 5 above)
Comment (AlexeySachkov) [reply SYCL/Regression/set-arg-local-accessor.cpp]: Done in 8cfb6f4c98e7f7d71f6d7f2da3d8bb37e0496759

File: SYCL/Regression/set-arg-local-accessor.cpp (hunk 4 above)
Comment (AlexeySachkov) [reply SYCL/Regression/set-arg-local-accessor.cpp]: There is no need for that, removed initialization in 8cfb6f4c98e7f7d71f6d7f2da3d8bb37e0496759

File: SYCL/Regression/set-arg-local-accessor.cpp (hunk 2 above)
Comment (AlexeySachkov) [reply SYCL/Regression/set-arg-local-accessor.cpp:18]: It is not needed indeed. Refactored kernel so the barrier is useful in 8cfb6f4c98e7f7d71f6d7f2da3d8bb37e0496759 and 3620717208f1d81ac6b0d29fd7c00ffbad0864b7

**Review by romanovvlad — APPROVED** [review]

**Pull request merged**